packages/eslint-plugin/src/rules/prefer-optional-chain.ts

```diff
@@ -109,7 +116,105 @@ export default createRule<
+          } & TSESTree.IfStatement,
+        ): void => {
+          if (!options.allowSuggestingOnIfStatements) {
+            return;
+          }
+          if (node.alternate) {
+            return;
+          }
+          const ifBodyStatement =
+            node.consequent.type === AST_NODE_TYPES.BlockStatement
+              ? node.consequent.body[0]
+              : node.consequent;
+
+          if (ifBodyStatement.type !== AST_NODE_TYPES.ExpressionStatement) {
+            return;
+          }
+
+          const hasCommentsInIfBody =
+            context.sourceCode.getCommentsBefore(ifBodyStatement).length ||
+            context.sourceCode.getCommentsAfter(ifBodyStatement).length;
+
+          if (hasCommentsInIfBody) {
+            return;
+          }
+          const ifBodyExpression = ifBodyStatement.expression;
+          const currentChain: ValidOperand[] = [
+            {
+              node: node.test,
+              type: OperandValidity.Valid,
+              comparedName: node.test,
+              comparisonType: NullishComparisonType.Boolean,
+              isYoda: false,
+            },
+          ];
+          if (node.test.type === AST_NODE_TYPES.LogicalExpression) {
+            const { newlySeenLogicals, operands } = gatherLogicalOperands(
+              node.test,
+              parserServices,
+              context.sourceCode,
+              options,
+            );
+            for (const logical of newlySeenLogicals) {
+              seenLogicals.add(logical);
+            }
+            for (const operand of operands) {
+              if (operand.type === OperandValidity.Invalid) {
+                return;
+              }
+              currentChain.push(operand);
+            }
+          }
+          if (ifBodyExpression.type === AST_NODE_TYPES.LogicalExpression) {
+            const { newlySeenLogicals, operands } = gatherLogicalOperands(
+              ifBodyExpression,
+              parserServices,
+              context.sourceCode,
+              options,
+            );
+            for (const logical of newlySeenLogicals) {
+              seenLogicals.add(logical);
+            }
+            for (const operand of operands) {
+              if (operand.type === OperandValidity.Invalid) {
+                return;
+              }
+              currentChain.push(operand);
+            }
+          } else {
+            if (ifBodyExpression.type !== AST_NODE_TYPES.CallExpression) {
+              return;
+            }
+            currentChain.push({
+              node: ifBodyExpression,
+              type: OperandValidity.Valid,
+              comparedName: ifBodyExpression,
+              comparisonType: NullishComparisonType.Boolean,
+              isYoda: false,
+            });
+          }
+          analyzeChain(
+            context,
+            parserServices,
+            options,
+            node,
+            '&&',
+            currentChain,
+          );
+          return;
```

Comment:
[Style] Nit: no need; it's at the end of a function anyway.

```suggestion
```

Reply:
Is there a reason `no-useless-return` is not enabled?

Reply:
Not that I know of 😄. It might just be that we never thought to look at it. I'd be in favor if you want to file a separate issue 😄